dll/win32/shell32/shlfolder.cpp

```diff
@@ -274,6 +274,41 @@ HRESULT SHELL32_CompareDetails(IShellFolder2* isf, LPARAM lParam, LPCITEMIDLIST
     return MAKE_COMPARE_HRESULT(ret);
 }
 
+HRESULT SHELL_CompareAllFields(IShellFolder* psf,
+                                 LPCITEMIDLIST pidl1, LPCITEMIDLIST pidl2,
+                                 UINT Count, int Skip)
+{
+    for (UINT i = 0; i < Count; ++i)
+    {
+        if (Skip == (int)i)
+            continue;
+        HRESULT hr = psf->CompareIDs(i, pidl1, pidl2);
+        if (hr && SUCCEEDED(hr))
+            return hr;
+    }
+    return MAKE_COMPARE_HRESULT(0);
+}
+
+HRESULT SHELL_FolderImplCompareIDsTiebreaker(IShellFolder2* psf, LPARAM lParam,
+                                             LPCITEMIDLIST pidl1, LPCITEMIDLIST pidl2,
+                                             UINT Count, int Canonical)
+{
+    if ((lParam & SHCIDS_COLUMNMASK) != Canonical && Canonical >= 0)
+    {
+        LPARAM flags = (lParam & SHCIDS_BITMASK) & ~SHCIDS_ALLFIELDS;
+        HRESULT hr = psf->CompareIDs(Canonical | flags, pidl1, pidl2);
+        if (hr && SUCCEEDED(hr))
+            return hr;
+    }
+    if (lParam & SHCIDS_ALLFIELDS)
+    {
+        HRESULT hr = SHELL_CompareAllFields(psf, pidl1, pidl2, Count, Canonical);
+        if (hr && SUCCEEDED(hr))
+            return hr;
+    }
+    return SHELL32_CompareChildren(psf, lParam, pidl1, pidl2);
```

Comment:
This has the potential to recursively search an entire drive?

Reply:
No, it might search a path like c:\foo\bar\file.ext but it will never be a wildcard search. It should not hit the disk either most of the time.

This call is already present in the old code...

Reply:
That being said, I'm a little unsure if we should be passing the column down. The column might not make sense in folders deeper down.

I'll change that unless somebody says otherwise...

Reply:
Do you have any new recent new insight about this hypothetical recursion problem?

Reply:
There is no recursion problem. `SHELL32_CompareChildren` (which CFSFolder.cpp was already calling in the same'ish manner in the same place) will walk a PIDL item by item if it has to, until it reaches the end.

If we ask the root IShellFolder to compare "c:\foo\bar\aaa.ext" vs "c:\foo\bar\bbb.ext" for example, the IShellFolder for "foo" might compare "bar" and "bar" to be the same and so it asks the IShellFolder for "bar" to compare "aaa.ext" and "bbb.ext". They don't match (if the caller wanted to compare names), or they might match (if comparing date, size, etc). Either way, this is  the end and it will not recurse deeper.